pkg/admin/prerun/prerun.go

```diff
@@ -288,22 +281,67 @@ func (pr *PreRun) restore() error {
 	)
 	backupsForDeployment := getExistingBackupsForTheDeployment(existingBackups, currentDeploymentID)
 
-	if len(backupsForDeployment) == 0 {
-		return fmt.Errorf("there is no backup to restore for current deployment %q", currentDeploymentID)
-	}
-
 	if len(backupsForDeployment) > 1 {
 		// could happen during backing up when removing older backups failed
 		klog.InfoS("TODO: more than 1 backup, need to pick most recent one")
 	}
 
-	err = pr.dataManager.Restore(backupsForDeployment[0])
+	if len(backupsForDeployment) > 0 {
+		err = pr.dataManager.Restore(backupsForDeployment[0])
+		if err != nil {
+			return fmt.Errorf("failed to restore backup: %w", err)
+		}
+		klog.Info("Finished handling unhealthy system")
+		return nil
+	}
+
+	klog.InfoS("There is no backup to restore for current deployment - trying to restore backup for rollback deployment")
+	rollbackDeployID, err := getRollbackDeploymentID()
 	if err != nil {
-		return fmt.Errorf("failed to restore backup: %w", err)
+		return err
 	}
 
-	klog.Info("Finished restore")
-	return nil
+	if rollbackDeployID == "" {
+		// No backup for current deployment and there is no rollback deployment.
+		// This could be a unhealthy system that was manually rebooted to
+		// remediate the situation - let's not interfere: no backup, no restore, just proceed.
+		klog.InfoS("System has no rollback but health.json suggests system was rebooted - exiting prerun and proceeding with start")
+		return nil
+	}
+
+	klog.InfoS("Obtained rollback deployment",
+		"rollback-deployment-id", rollbackDeployID,
+		"current-deployment-id", currentDeploymentID,
+		"health-deployment-id", health.DeploymentID)
+
+	if health.DeploymentID == rollbackDeployID {
+		return fmt.Errorf("deployment ID stored in health.json is the same as rollback's" +
+			" - MicroShift should not be updated from unhealthy system")
+	}
+
+	if health.DeploymentID == currentDeploymentID {
+		// There is no backup for current deployment, but there is a backup for the rollback.
+		// Let's restore it and act like it's first boot of newly staged deployment
+		backupsForRollback := getExistingBackupsForTheDeployment(existingBackups, rollbackDeployID)
+		if len(backupsForRollback) == 0 {
+			// This could happen if current deployment is unhealthy and rollback didn't run MicroShift
+			klog.InfoS("There is no backup for rollback deployment as well - removing existing data for clean start")
+			return pr.dataManager.RemoveData()
+		}
+
+		klog.InfoS("Restoring backup for a rollback deployment to perform migration and try starting again",
```

Comment:
We don't seem to _know_ that there is a backup for the rollback until this point, so maybe the comment on line 323 is anticipating reaching this point?